src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+            end
+        end
+    end
+
+    # Fill in default values if none exits
+    if !haskey(res, "year") && !haskey(res, "month") && !haskey(res, "day")
+        res["year"], res["month"], res["day"] = Dates.yearmonthday(default)
+    else
+        if haskey(res, "year")
+            res["year"] = convertyear(res["year"])
+        else
+            res["year"] = year(default)
+        end
+        if !haskey(res, "month")
+            res["month"] = 1
+        end
+        if !haskey(res, "day")
+            res["day"] = 1
+        end
+    end
+    if !haskey(res, "hour") && !haskey(res, "minute") && !haskey(res, "second") && !haskey(res, "millisecond")
+        res["hour"] = hour(default)
+        res["minute"] = minute(default)
+        res["second"] = second(default)
+        res["millisecond"] = millisecond(default)
+    else
+        if !haskey(res, "hour")
+            res["hour"] = 0
+        end
+        if !haskey(res, "minute")
+            res["minute"] = 0
+        end
+        if !haskey(res, "second")
+            res["second"] = 0
+        end
+        if !haskey(res, "millisecond")
+            res["millisecond"] = 0
+        end
+    end
+    # determine timezone
+    if !haskey(res, "tzname") && !haskey(res, "tzoffset")
+        res["timezone"] = default.timezone
+    elseif !haskey(res, "tzoffset")
+        if haskey(timezone_infos, res["tzname"])
+            res["timezone"] = timezone_infos[res["tzname"]]
+        elseif res["tzname"] in TimeZones.timezone_names()
+            res["timezone"] = TimeZone(res["tzname"])
+        elseif lowercase(res["tzname"]) in UTCZONE
+            res["timezone"] = TimeZone("utc")
+        else
+            error("Failed to parse date")
+        end
+    elseif !haskey(res, "tzname")
+        res["timezone"] = FixedTimeZone("local",res["tzoffset"])
+    else
+        res["timezone"] = FixedTimeZone(res["tzname"], res["tzoffset"])
+    end
+
+    return ZonedDateTime(DateTime(res["year"], res["month"], res["day"], res["hour"],
+            res["minute"], res["second"], res["millisecond"]), res["timezone"])
+end
+
+# converts a 2 digit year to a 4 didgit one near year 2000 (e.g. 95 becomes 1995)
+function convertyear(year::Int)
+    if year > 99
+        return year
+    elseif year < 50
+        return year + 2000
+    else
+        return year + 1900
+    end
+end
+
+function _parsedatetokens(input::AbstractString)
+    tokens = []
+    regex = r"^(?P<token>(\d+\.\d+(?=[^\.\d]|$))|(\d+)|(((?=[^\d])\w)+))(?P<extra>.*)$"
+    input = strip(input)
+    while !isempty(input)
+        if ismatch(regex,input)
+            tokenmatch = match(regex, input)
+            push!(tokens, tokenmatch["token"])
+            input = tokenmatch["extra"]
+        else
+            if ismatch(r"\s", string(input[1:1]))
+                push!(tokens, " ")
+            else
+                push!(tokens, input[1:1])
+            end
+            input = strip(input[2:end])
```

Comment:
Definitely not `UTF8String` compatible. eg. `DateTimeParser._parsedatetokens("∀∃")`